lnst/Recipes/ENRT/SRIOVNetnsOvSRecipe.py

```diff
@@ -0,0 +1,197 @@
+        dict(gro="on", gso="on", tso="on", tx="on", rx="on"),
+        dict(gro="off", gso="on", tso="on", tx="on", rx="on"),
+        dict(gro="on", gso="off", tso="off", tx="on", rx="on"),
+        dict(gro="on", gso="on", tso="off", tx="off", rx="on"),
+        dict(gro="on", gso="on", tso="on", tx="on", rx="off")))
+
+    def test_wide_configuration(self):
+        """
+        Test wide configuration for this recipe involves switching the device to switchdev
+        mode, adding single VF and mapping the VF, as well as its representor.
+        New namespace is created to mimic container networking, where the VF is moved.
+        Next, VF is assigned an IPv4 and IPv6 address on both hosts.
+        Finally, new OvS bridge is created, where PF and VF representor are added, to
+        ensure correct switching of network traffic between the hosts.
+
+        host1.eth0 = 192.168.101.1/24 and fc00::1/64
+
+        host2.eth0 = 192.168.101.2/24 and fc00::2/64
+        """
+        host1, host2 = self.matched.host1, self.matched.host2
+        configuration = super().test_wide_configuration()
+        configuration.test_wide_devices = []
+
+        for i, host in enumerate([host1, host2]):
+            host.run("systemctl enable openvswitch")
+            host.run("systemctl start openvswitch")
+            host.run("ovs-vsctl set Open_vSwitch . other_config:hw-offload=true")
+
+            host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode switchdev; sleep 2")
+            host.run(f"echo 1 > /sys/class/net/{host.eth0.name}/device/sriov_numvfs; sleep 5")
+
+            vf_ifname = dict(ifname=f"{host.eth0.name}v0")
+            host.map_device("vf_eth0", vf_ifname)
+
+            host.newns = NetNamespace(f"lnst{i + 1}")
+            host.newns.vf_eth0 = host.vf_eth0
+
+            vf_representor_ifname = dict(ifname="eth0")
+            host.map_device("vf_representor_eth0", vf_representor_ifname)
+            host.vf_representor = host.vf_representor_eth0
+
+            host.br0 = OvsBridgeDevice()
+            host.br0.port_add(host.eth0)
+            host.br0.port_add(host.vf_representor)
+
+            for dev in [host.eth0, host.newns.vf_eth0, host.vf_representor, host.br0]:
+                dev.up()
+
+            host.newns.vf_eth0.ip_add(ipaddress("192.168.101." + str(i + 1) + "/24"))
+            host.newns.vf_eth0.ip_add(ipaddress("fc00::" + str(i + 1) + "/64"))
+
+            configuration.test_wide_devices.append(host.newns.vf_eth0)
+
+        self.wait_tentative_ips(configuration.test_wide_devices)
+
+        return configuration
+
+    def generate_test_wide_description(self, config):
+        """
+        Test wide description consists of mirror setup of a single namespace,
+        containing virtual function. Representor of this virutal function, as well as,
+        physical function is plugged into the OvS bridge, which has hardware offloading
+        capabilities enabled.
+        """
```

Comment:
I'm not sure if this docstring is useful in th context of this method. Probably makes sense to include this in the top level class docstring or the `test_wide_configuraiton` docstring.

Reply:
Removed